File: 102 Binary Tree Level Order Traversal/102.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from collections import deque
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        res = []
        if not root:
            return res
        q = deque()
        q.append(root)
        while q:
            size = len(q)
            item = []
            for i in range(size):
                node = q.popleft()
                item.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            res.append(item)
        return res

File: 102 Binary Tree Level Order Traversal/102.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let res = []
    if (!root) {
        return res;
    }
    let queue = []
    queue.push(root)
    while (queue.length) {
        let size = queue.length
        let item = []
        for (let i = 0; i < size; ++i) {
            let node = queue.shift()
            item.push(node.val)
            if (node.left) {
                queue.push(node.left)
            }
            if (node.right) {
                queue.push(node.right)
            }
        }
        res.push(item)
    }
    return res;
};